packages/common/src/extensions/__tests__/slickRowBasedEdit.spec.ts

```diff
@@ -0,0 +1,824 @@
+
+type MockedSlickGrid = SlickGrid & {
+  [K in keyof SlickGrid]: jest.Mock<any, any>;
+};
+
+describe('Row Based Edit Plugin', () => {
+  let plugin: SlickRowBasedEdit;
+  let gridStub: MockedSlickGrid;
+  let gridService: GridService;
+
+  beforeEach(() => {
+    const _any = {} as any;
+    gridStub = {
+      ...(gridStubBlueprint as unknown as SlickGrid),
+      columns: [...mockColumns],
+    } as unknown as MockedSlickGrid;
+    gridService = new GridService(_any, _any, _any, _any, _any, _any, _any);
+    jest
+      .spyOn(gridService, 'getAllColumnDefinitions')
+      .mockReturnValue(mockColumns);
+    plugin = new SlickRowBasedEdit(pubSubServiceStub, addonOptions);
+    (plugin as any)._eventHandler = {
+      subscribe: jest.fn(),
+      unsubscribeAll: jest.fn(),
+    };
+
+    // add a getter eventHandler to the gridStub
+    Object.defineProperty(gridStub, 'eventHandler', {
+      get: jest.fn(() => (plugin as any)._eventHandler),
+      set: jest.fn(),
+      enumerable: true,
+      configurable: true,
+    });
+  });
+
+  afterEach(() => {
+    plugin.destroy();
+    plugin.dispose();
+  });
+
+  it('should create the plugin', () => {
+    expect(plugin).toBeTruthy();
+    expect(plugin.eventHandler).toBeTruthy();
+  });
+
+  const optionsMock = {
+    autoEdit: true,
+    editable: true,
+    enableCellNavigation: true,
+  } as GridOption;
+
+  it('should only allow cell editing when the currently edited row is in edit mode', () => {
+    const fakeItem = { id: 'test' };
+
+    gridStub.getData.mockReturnValue({ getItem: () => fakeItem });
+    gridStub.getOptions.mockReturnValue(optionsMock);
+
+    plugin.init(gridStub, gridService);
+
+    const onBeforeEditCellHandler = (
+      plugin.eventHandler.subscribe as jest.Mock<any, any>
+    ).mock.calls[0][1] as (
+      e: Event,
+      args: OnBeforeEditCellEventArgs
+    ) => boolean;
+    expect(
+      onBeforeEditCellHandler(
+        {} as Event,
+        { item: fakeItem } as OnBeforeEditCellEventArgs
+      )
+    ).toBe(false);
+
+    plugin.rowBasedEditCommandHandler(
+      fakeItem,
+      {} as Column,
+      {} as EditCommand
+    );
+
+    expect(
+      onBeforeEditCellHandler(
+        {} as Event,
+        { item: fakeItem } as OnBeforeEditCellEventArgs
+      )
+    ).toBe(true);
+  });
+
+  it('should throw an error when "enableRowEdit" is set without "enableCellNavigation"', () => {
+    gridStub.getOptions.mockReturnValue({});
+
+    expect(() => plugin.init(gridStub, gridService)).toThrow(
+      /(enableCellNavigation = true)/
+    );
+  });
+
+  it('should throw an error when "enableRowEdit" is set without "editable"', () => {
+    gridStub.getOptions.mockReturnValue({ enableCellNavigation: true });
+
+    expect(() => plugin.init(gridStub, gridService)).toThrow(
+      /(editable = true)/
+    );
+  });
+
+  it('should warn the user when autoEdit is not set or false and turn it on', () => {
+    const consoleSpy = jest.spyOn(console, 'warn');
```

Comment:
I just saw in the GitHub Action that this console is showing up, to avoid showing them in Jest, you can (should) use the following
```ts
const consoleSpy = jest.spyOn(global.console, 'warn').mockReturnValue();
```
![image](https://github.com/ghiscoding/slickgrid-universal/assets/643976/fafe8207-22e1-496c-ac18-e41daf98e010)